src/game/client/worldheightmap.cpp

```diff
@@ -29,21 +45,2359 @@ class GDIFileStream : public InputStream
+}
+
+bool WorldHeightMap::Parse_Height_Map_Data_Chunk(DataChunkInput &file, DataChunkInfo *info, void *user_data)
+{
+    return ((WorldHeightMap *)user_data)->Parse_Height_Map_Data(file, info, user_data);
+}
+
+bool WorldHeightMap::Parse_Height_Map_Data(DataChunkInput &file, DataChunkInfo *info, void *user_data)
+{
+    m_width = file.Read_Int32();
+    m_height = file.Read_Int32();
+
+    if (info->version < 3) {
+        m_borderSize = 0;
+    } else {
+        m_borderSize = file.Read_Int32();
+    }
+
+    if (info->version < 4) {
+        m_boundaries.resize(1);
+        m_boundaries[0].x = m_width - 2 * m_borderSize;
+        m_boundaries[0].y = m_height - 2 * m_borderSize;
+    } else {
+        int size = file.Read_Int32();
+        m_boundaries.resize(size);
+
+        for (int i = 0; i < size; i++) {
+            m_boundaries[i].x = file.Read_Int32();
+            m_boundaries[i].y = file.Read_Int32();
+        }
+    }
+
+    m_dataSize = file.Read_Int32();
+    m_data = new unsigned char[m_dataSize];
+
+    if (m_dataSize <= 0 || m_dataSize != m_height * m_width) {
+        throw CODE_05;
+    }
+
+    m_seismicUpdateWidth = (m_width + 7) / 8;
+    m_seismicUpdateFlag = new unsigned char[m_height * m_seismicUpdateWidth];
+    Clear_Seismic_Update_Flags();
+    m_seismicZVelocities = new float[m_dataSize];
+    Fill_Seismic_Z_Velocities(0.0f);
+    file.Read_Byte_Array(m_data, m_dataSize);
+
+    if (info->version == 1) {
+        int x = (m_width + 1) / 2;
+        int y = (m_height + 1) / 2;
+
+        for (int i = 0; i < y; i++) {
+            for (int j = 0; j < x; j++) {
+                m_data[j + y * i] = m_data[2 * j + m_width * 2 * i];
+            }
+        }
+    }
+
+    captainslog_dbgassert(file.At_End_Of_Chunk(), "Unexpected data left over.");
+    return true;
+}
+
+bool WorldHeightMap::Parse_Size_Only_In_Chunk(DataChunkInput &file, DataChunkInfo *info, void *user_data)
+{
+    return ((WorldHeightMap *)user_data)->Parse_Size_Only(file, info, user_data);
+}
+
+bool WorldHeightMap::Parse_Size_Only(DataChunkInput &file, DataChunkInfo *info, void *user_data)
+{
+    m_width = file.Read_Int32();
+    m_height = file.Read_Int32();
+
+    if (info->version < 3) {
+        m_borderSize = 0;
+    } else {
+        m_borderSize = file.Read_Int32();
+    }
+
+    if (info->version < 4) {
+        m_boundaries.resize(1);
+        m_boundaries[0].x = m_width - 2 * m_borderSize;
+        m_boundaries[0].y = m_height - 2 * m_borderSize;
+    } else {
+        int size = file.Read_Int32();
+        m_boundaries.resize(size);
+
+        for (int i = 0; i < size; i++) {
+            m_boundaries[i].x = file.Read_Int32();
+            m_boundaries[i].y = file.Read_Int32();
+        }
+    }
+
+    m_dataSize = file.Read_Int32();
+    m_data = new unsigned char[m_dataSize];
+
+    if (m_dataSize <= 0 || m_dataSize != m_height * m_width) {
+        throw CODE_05;
+    }
+
+    m_seismicUpdateWidth = (m_width + 7) / 8;

```

Comment:
seismic shouldn't be here as this parses size only

Reply:
fixed.